lib/internal/vm/module.js

```diff
@@ -379,8 +379,17 @@ class SyntheticModule extends Module {
   constructor(exportNames, evaluateCallback, options = {}) {
     if (!ArrayIsArray(exportNames) ||
       exportNames.some((e) => typeof e !== 'string')) {
-      throw new ERR_INVALID_ARG_TYPE('exportNames', 'Array of strings',
+      throw new ERR_INVALID_ARG_TYPE('exportNames',
+                                     'Array of unique strings',
                                      exportNames);
+    } else {
+      exportNames.forEach((name, i) => {
+        if (exportNames.indexOf(name, i + 1) !== -1) {
+          throw new ERR_INVALID_ARG_TYPE('exportNames',
+                                         'Array of unique strings',
+                                         exportNames);
```

Comment:
the _type_ here isn't wrong, only its values. Would something like `ERR_INVALID_ARG_VALUE` make more sense?